spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -565,10 +561,88 @@ def _measure_tissue_bridges(self, im_lesion_data, p_lst, idx):
+
+        self.measure_pd.loc[idx, 'length_midsagittal_slice [mm]'] = length_cur
+        printv(f'  (S-I) length in the midsagittal slice: {(np.round(length_cur, 2))} mm',
+               self.verbose, type='info')
+
+    def _measure_width_midsagittal_slice(self, im_lesion_data, p_lst, idx):
+        """
+        Measure the width of the lesion along the anterior-posterior axis in the **midsagittal slice** when taking into
+        account the angle correction.
+        The width is defined as the maximum lesion width in the A-P axis across all axial slices with the lesion in
+        the midsagittal slice.
+
+        :param im_lesion_data: 3D numpy array: mask of the lesion. The orientation is assumed to be RPI (because we
+        reoriented the image to RPI using orient2rpi())
+        :param p_lst: list, pixel size of the lesion
+        :param idx: int, index of the lesion
+        """
+        # Fetch a list of axial slice numbers that are nonzero in the mid-sagittal slice (RPI)
+        im_data_midsagittal = im_lesion_data[self.midsagittal_sc_slice_rpi, :, :]  # 3D -> 2D, dim=[AP, SI]
+        nonzero_axial_slices = np.unique(np.where(im_data_midsagittal)[1])  # [1] -> SI
+
+        # Iterate across axial slices to compute lesion width
+        lesion_width_dict = {}
+        for axial_slice in nonzero_axial_slices:
+            # Get the lesion segmentation mask of the selected 2D axial slice
+            slice_lesion_data = im_lesion_data[self.midsagittal_sc_slice_rpi, :, axial_slice]
+            # Get the indices of the lesion mask for the selected axial slice to compute the lesion width.
+            # The lesion width is defined as max - min + 1
+            # Note: we intentionally use 'max - min + 1' instead of 'len(np.where(slice_lesion_data)[0])' because the
+            # 'len' approach would return the number of elements, which could be influenced, for example, by the
+            # presence of holes in the lesion mask.
+            # Context: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4617#discussion_r1744031056
+            slice_min = np.where(slice_lesion_data)[0][0]     # [0] returns the most dorsal elements
+            slice_max = np.where(slice_lesion_data)[0][-1]    # [-1] returns the most ventral elements
+            lesion_width_dict[axial_slice] = slice_max - slice_min + 1
+
+        # Get the width in mm (apply the angle correction)
+        width_cur_dict = {axial_slice: p_lst[1] * np.cos(self.angles_sagittal[axial_slice]) * lesion_width
+                          for axial_slice, lesion_width in lesion_width_dict.items()}
+
+        # Get the maximum width across all axial slices
+        # Check if width_cur_dict is empty, if so, it means that the lesion does not exist in the midsagittal slice.
+        # In this case, set the width to 'np.nan'
+        if not width_cur_dict:
+            width_cur = np.nan
+        else:
+            width_cur = max(width_cur_dict.values())
```

Comment:
Just a tiny tiny nitpick:

```python
width_cur =  max(width_cur_dict.values()) if width_cur_dict else np.nan
```

Reply:
Nice! Done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4617/commits/266e1922e6fb8d623e9a3e33f647e2a0caecc658